scripts/tools/metricExportData.yml

```diff
@@ -0,0 +1,129 @@
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Export Metrics Data for system independent calculation."
+version: "1.6"
+class: $tMetricGroup
+name: GetExportDataExp
+decl: static
+details:
+    - "This function exports raw data and necessary information to perform metrics calculation of collected data in a different system than where data was collected, which may or may not have accelerators."
+    - "Implementations can choose to describe the data arrangement of the exported data, using any mechanism which allows users to read and process them."
+    - "The application may call this function from simultaneous threads."
+params:
+    - type: "$t_metric_group_handle_t"
+      name: hMetricGroup
+      desc: "[in] handle of the metric group"
+    - type: "const uint8_t*"
+      name: pRawData
+      desc: "[in] buffer of raw data"
+    - type: size_t
+      name: rawDataSize
+      desc: "[in] size in bytes of raw data buffer"
+    - type: "size_t*"
+      name: pExportDataSize
+      desc: |
+            [in,out] size in bytes of export data buffer
+            if size is zero, then the driver shall update the value with the number of bytes necessary to store the exported data.
+            if size is greater than required, then the driver shall update the value with the actual number of bytes necessary to store the exported data.
+    - type: "uint8_t *"
+      name: pExportData
+      desc: |
+            [in,out][optional][range(0, *pExportDataSize)] buffer of exported data.
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Calculate one or more sets of metric values from exported raw data."
+version: "1.6"
+class: $xDriver
+name: CalculateMetricExportDataExp
+decl: static
+details:
+    - "Calculate metrics values using exported data returned by $tMetricGroupGetExportDataExp."
+    - "This function is similar to $tMetricGroupCalculateMultipleMetricValuesExp except it would calculate from exported metric data."
+    - "This function could be used to calculate metrics on a system different from where the metric raw data was collected."
+    - "The application may call this function from simultaneous threads."
+params:
+    - type: $x_driver_handle_t
+      name: hDriver
+      desc: "[in] handle of the driver instance"
+    - type: "$t_metric_group_calculation_type_t"
+      name: type
+      desc: "[in] calculation type to be applied on raw data"
+    - type: size_t
+      name: exportDataSize
+      desc: "[in] size in bytes of exported data buffer"
+    - type: "const uint8_t*"
+      name: pExportData
+      desc: "[in][range(0, exportDataSize)] buffer of exported data to calculate"
+    - type: "$t_metric_calculate_exp_desc_t*"
+      name: pCalculateDescriptor
+      desc: "[in] descriptor specifying calculation specific parameters"
+    - type: uint32_t*
+      name: pSetCount
```

Comment:
these two pCount are not marked as optional, so they must be passed always, at the same time?

Reply:
Yes.. 
We need to query for the 
* number of sets
* total number of metrics available

So that application can allocate accordingly